### ISSUE TITLE
[fltk] <fltk>/<1.3.8>: fails to link to libxft on Linux

### ISSUE THREAD

**Issue by erikogenvik** [conan-io/conan-center-index#15276]
### Description

When building on Linux it fails to link to required libxft. Symbols are missing, like this:
```
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: /home/erik/.conan/data/fltk/1.3.8/_/_/package/7fb45cb79434bbfd311c6fcceb81f069245765ae/lib/libfltk.a(fl_font.cxx.o): in function `Fl_Font_Descriptor::Fl_Font_Descriptor(char const*, int, int)':
fl_font.cxx:(.text+0x469): undefined reference to `XftFontOpenXlfd'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: fl_font.cxx:(.text+0x6b7): undefined reference to `XftFontMatch'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: fl_font.cxx:(.text+0x6c8): undefined reference to `XftFontOpenPattern'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: fl_font.cxx:(.text+0x715): undefined reference to `XftFontOpen'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: /home/erik/.conan/data/fltk/1.3.8/_/_/package/7fb45cb79434bbfd311c6fcceb81f069245765ae/lib/libfltk.a(fl_font.cxx.o): in function `Fl_Xlib_Graphics_Driver::width(char const*, int)':
fl_font.cxx:(.text+0xb9c): undefined reference to `XftTextExtents32'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: /home/erik/.conan/data/fltk/1.3.8/_/_/package/7fb45cb79434bbfd311c6fcceb81f069245765ae/lib/libfltk.a(fl_font.cxx.o): in function `Fl_Xlib_Graphics_Driver::width(unsigned int)':
fl_font.cxx:(.text+0xc82): undefined reference to `XftTextExtents32'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: /home/erik/.conan/data/fltk/1.3.8/_/_/package/7fb45cb79434bbfd311c6fcceb81f069245765ae/lib/libfltk.a(fl_font.cxx.o): in function `Fl_Xlib_Graphics_Driver::text_extents(char const*, int, int&, int&, int&, int&)':
fl_font.cxx:(.text+0xd0b): undefined reference to `XftTextExtents32'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: /home/erik/.conan/data/fltk/1.3.8/_/_/package/7fb45cb79434bbfd311c6fcceb81f069245765ae/lib/libfltk.a(fl_font.cxx.o): in function `Fl_Xlib_Graphics_Driver::draw(char const*, int, int, int)':
fl_font.cxx:(.text+0x123c): undefined reference to `XftDrawChange'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: fl_font.cxx:(.text+0x1272): undefined reference to `XftDrawSetClip'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: fl_font.cxx:(.text+0x1309): undefined reference to `XftDrawString32'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: fl_font.cxx:(.text+0x13b9): undefined reference to `XftDrawCreate'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: /home/erik/.conan/data/fltk/1.3.8/_/_/package/7fb45cb79434bbfd311c6fcceb81f069245765ae/lib/libfltk.a(fl_font.cxx.o): in function `Fl_Xlib_Graphics_Driver::rtl_draw(char const*, int, int, int)':
fl_font.cxx:(.text+0x1571): undefined reference to `XftTextExtents32'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: fl_font.cxx:(.text+0x15a2): undefined reference to `XftDrawChange'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: fl_font.cxx:(.text+0x15d8): undefined reference to `XftDrawSetClip'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: fl_font.cxx:(.text+0x166d): undefined reference to `XftDrawString32'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: fl_font.cxx:(.text+0x16b9): undefined reference to `XftDrawCreate'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: /home/erik/.conan/data/fltk/1.3.8/_/_/package/7fb45cb79434bbfd311c6fcceb81f069245765ae/lib/libfltk.a(fl_font.cxx.o): in function `fl_destroy_xft_draw(unsigned long)':
fl_font.cxx:(.text+0x11e9): undefined reference to `XftDrawChange'
```

Fix is to include libxft in requirements, which I'll do in a separate PR.

### Package and Environment Details

* Package Name/Version: **fltk/1.3.8**
* Operating System+version: **OpenSuSE Tumbleweed**
* Compiler+version: **GCC 12.2.1**
* Conan version: **conan 1.56.0**
* Python version: **Python 3.10.9**

### Conan profile

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=12.2
compiler.libcxx=libstdc++11
build_type=Release
[options]
[conf]
[build_requires]
[env]

### Steps to reproduce

conan create all/conanfile.py fltk/1.3.8@ -pr:b=default -pr:h=default

### Logs

<details><summary>Click to expand log</summary>

```
[log.zip](https://github.com/conan-io/conan-center-index/files/10420333/log.zip)
```

</details>

**Comment by paulaltin** [comment]
@jputcu – I'm having trouble getting FLTK to use Xft on Linux. I've tried setting the `with_xft` option to `True` in my project's conanfile (this is with Conan version 1.57.0), but it gives me the following error:

```
ERROR: fltk/1.3.8: option 'with_xft' doesn't exist
Possible options are ['shared', 'fPIC', 'with_gl', 'with_threads', 'abi_version']
```

Your comment above is the only mention I can find of this option, so I thought I'd ask here before starting a new thread..

Thanks in advance!

**Comment by jputcu** [comment]
@paulaltin I've tried it with the newest conan2, so maybe that is the reason.

**Comment by paulaltin** [comment]
@jputcu – I was able to get it working with conan1 by requesting a different recipe revision (`fd5ef724f4cd9281811253e8e1f91a1b` instead of `3fa62825aefcc00ae634fab447e5c09d`). I'm not sure why the latter is still shown as the latest revision on [conan.io](https://conan.io/center/recipes/fltk?version=1.3.8).

Anyway, thanks for your help, and for adding this option in #20558.

**Comment by uilianries** [comment]
The version 1.3.8 is no longer updated in Conan Center (but still available). To keep updated, please, use the latest version available: **1.4.4**:

    conan install -r conancenter --update --requires=fltk/1.4.4 -o "&:with_xft=True"

The fltk-1.4.4 can be built with libxft without errors on Linux:

* Fltk 1.4.4 + libxft + Linux + amd64 + gcc13 + all shared: [fltk-1.4.4-linux-amd64-gcc13-release-shared-xft.log](https://github.com/user-attachments/files/24887331/fltk-1.4.4-linux-amd64-gcc13-release-shared-xft.log)
* Fltk 1.4.4 + libxft + Linux + amd64 + gcc13 + all static: [fltk-1.4.4-linux-amd64-gcc13-release-static-xft.log](https://github.com/user-attachments/files/24887332/fltk-1.4.4-linux-amd64-gcc13-release-static-xft.log)


I'll be closing this issue as a newer version available in Conan Center does not present the commented error. Please, feel free to give your feedback. Regards.
